server.cpp

```diff
@@ -70,8 +140,32 @@ void write_response(tcp::socket& socket, const Request& request, const std::stri
     }
     else
     {
-        const std::string message = "HTTP/1.1 200 OK\r\nHost: localhost\r\n\r\n" + date;
-        boost::asio::write(socket, boost::asio::buffer(message), ignored_error);
+        std::string path;
+
+        if (!work_dir.empty())
+            path = work_dir;
+        else
+            path = ".";
+
+        DIR *dir = opendir(path.c_str());
```

Comment:
You only need `dir` when you send index page, but you open it event if a file is requested.